samples/bluetooth/direction_finding_connectionless_rx/src/main.c

```diff
@@ -358,17 +365,26 @@ void main(void)
 
 		create_sync();
 
+		sync_wait = true;
+		sync_terminated = false;
+
 		printk("Waiting for periodic sync...\n");
 		err = k_sem_take(&sem_per_sync, K_MSEC(sync_create_timeout_ms));
-		if (err != 0) {
-			printk("failed (err %d)\n", err);
+		if (err != 0 || sync_terminated) {
+			if (err != 0) {
+				printk("failed (err %d)\n", err);
+			} else {
+				printk("terminated\n");
+			}
 			err = delete_sync();
 			if (err != 0) {
 				return;
 			}
+			sync_wait = false;
```

Comment:
Does it make sense to have this before the `delete_sync()`, in case that fails for whatever reason?

Reply:
I've moved it but in current state there is no difference because if `delete_sync` fails `main` will `return;`. I've moved it in case someone wants to make different handling of the `delete_sync` failures.